modules/common/condition/funcs.go

```diff
@@ -218,3 +231,111 @@ func CreateList(conditions ...*Condition) Conditions {
 	}
 	return cs
 }
+
+// Mirror - mirros the latest condition in a sorted conditionGroup into a target
+// condition of type t. The conditionGroup entries is split by Status with the
+// oder False, True, Unknown. If Status=False its again split into Severity by
+// Error, Warning, Info.
+func (conditions *Conditions) Mirror(t Type) *Condition {
```

Comment:
It looks like `Type` is never considered when finding the condition to mirror.  Just wondering if someone called this with `Type` `X` and if it matched against a condition with `Type` `Y`, if we might see unexpected behavior (i.e. you would return a `Type` `X` with `Reason`, `Severity` and/or `Message` belonging to a condition that was actually `Type` `Y`).  Or perhaps that is not even possible and I am misreading something.

Reply:
t Type is the target condition you want to mirror to. It is not expected that this type is in the conditions list you want to mirror from. mirror gets the latest condition from the list using the in the comment described logic on what gets prioritized, no matter if the type matches. The use case is you have a new overall condition type from a service and mirror the latest state on this from the actual service to the level up.

Reply:
maybe we should call it tt, or targetType?

Reply:
Ah, I think I get it now.  So the condition of the level up will be whatever `Type` it already is, but its `Message` (and maybe `Reason` and `Severity`) is to be determined from the condition that the `Mirror` function picks.

I guess what I found confusing was the statement `mirrors the latest condition...into a target condition of type t`, being that conditions contain a `Type`, so mirroring a condition made me think everything about the mirrored condition should be copied (including `Type`).  But yeah, if we're explicitly passing a "target" `Type`, it makes sense that it would take precedence.

Reply:
right that comment is confusing. I'll try to get a better comment there.

Reply:
updated the comment and rebased. I hope the new comment makes it more clear on what it does or does not